jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/CachingHttpContentFactory.java

```diff
@@ -403,6 +388,20 @@ public String getKey()
             return _cacheKey;
         }
 
+        @Override
+        public void writeTo(Content.Sink sink, long offset, long length, Callback callback)
+        {
+            try
+            {
+                sink.write(false, BufferUtil.slice(_buffer.getByteBuffer(), (int)offset, (int)length), callback);
```

Comment:
Why is this not last?
There are significant efficiencies to be had by sending content with last==true, rather than sending the content and then an empty buffer with last==true.

If you need a writeTo without last==true for multi-part ranges, then I think you need another method taking the boolean (or always pass the boolean).

Reply:
You just spotted a mistake: the last flag should be set to `true` here (and in the FileMapping impls too).

Multi-part ranges already deal with this by manipulating the last flag, see the comment I added in the [ee9 ResourceService](https://github.com/jetty/jetty.project/blob/219d442b3d4904f4c2f784434558482ba9e394ba/jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ResourceService.java#L868-L873).